relay-server/src/services/processor.rs

```diff
@@ -1229,44 +1207,47 @@ impl EnvelopeProcessorService {
         &self,
         state: &mut ProcessEnvelopeState,
         managed_envelope: &mut TypedEnvelope<Group>,
+        event: Annotated<Event>,
         project_info: Arc<ProjectInfo>,
         rate_limits: Arc<RateLimits>,
-    ) -> Result<(), ProcessingError> {
+    ) -> Result<Annotated<Event>, ProcessingError> {
         let global_config = self.inner.global_config.current();
         let rate_limiter = match self.inner.rate_limiter.as_ref() {
             Some(rate_limiter) => rate_limiter,
-            None => return Ok(()),
+            None => return Ok(event),
         };
 
         // Cached quotas first, they are quick to evaluate and some quotas (indexed) are not
         // applied in the fast path, all cached quotas can be applied here.
-        let _ = RateLimiter::Cached.enforce(
+        let cached_result = RateLimiter::Cached.enforce(
             state,
             managed_envelope,
+            event,
             &global_config,
             project_info.clone(),
             rate_limits.clone(),
         )?;
 
         // Enforce all quotas consistently with Redis.
-        let limits = RateLimiter::Consistent(rate_limiter).enforce(
+        let consistent_result = RateLimiter::Consistent(rate_limiter).enforce(
             state,
             managed_envelope,
+            cached_result.event,
```

Comment:
We supply the event downstream, since the modified state was doing this implicitly.